src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -595,39 +594,335 @@ public override bool IsTupleType
             }
         }
 
-        internal sealed override ObsoleteAttributeData ObsoleteAttributeData
+        public override bool IsImplicitlyDeclared
         {
             get
             {
-                return null;
+                return false;
+            }
+        }
+
+        public override ImmutableArray<TypeSymbol> TupleElementTypes
+        {
+            get
+            {
+                return _elementTypes;
+            }
+        }
+
+        public override ImmutableArray<string> TupleElementNames
+        {
+            get
+            {
+                return _elementNames;
             }
         }
 
         public override ImmutableArray<Symbol> GetMembers()
```

Comment:
I'm suggesting something like "Returns the methods of the underlying tuple type, together with the default named fields, and the custom named fields (if names are specified)".